source/rules/no-ignored-subscription.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { Rule } from "eslint";
+import * as es from "estree";
+import * as ts from "typescript";
+import { couldBeType } from "tsutils-etc";
+import { getTypeCheckerAndNodeMap } from "../utils";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description: "Disallows ignoring the subscription returned by subscribe.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Ignoring returned subscriptions is forbidden."
+    },
+    schema: []
+  },
+  create: context => {
+    const { nodeMap, typeChecker } = getTypeCheckerAndNodeMap(context);
+
+    return {
+      "ExpressionStatement[expression.callee.property.name='subscribe'] > CallExpression > MemberExpression": (
```

Comment:
Is there a particular reason the selector puts attribute specifier on the left-most type? And not on the right-most, like this?

```
ExpressionStatement > CallExpression > MemberExpression[property.name="subscribe"]
```

Reply:
IMO, specifying the attribute on the right-most AST type will be more efficient, as the implementation's call to the `split` method won't be made for every single `ExpressionStatement` that's traversed:

https://github.com/estools/esquery/blob/1853a6fbaf40709dc2bf9f3c027163a573c3824f/esquery.js#L19-L26

Reply:
Thanks for the tip. 
There isn't a reason, it was because the selector couldn't find the node with astexplorer.net previously.  Maybe it was because I made a mistake with selector or made a typo...

Reply:
Thanks. I really like the selector-based listeners. They make things so much simpler.

BTW, I discovered, today, that you can specify numeric indices in attribute selectors, like this:

```
[arguments.0.name='Boolean']
```

Maybe you already know this. I didn't. I only found it by poking around in the `esquery` issues.

Reply:
I also really like them, was a bit worried I couldn't used TSQuery anymore but then saw that eslint has esquery.

I did not know you could do that, thanks for mentioning this!
I tried `[arguments[0].name='Boolean']` before and I was bummed it didn't work.